### ISSUE TITLE
Remove components build-time dependencies from builder and react-sdk 

### ISSUE THREAD

**Issue by istarkov** [webstudio-is/webstudio-builder#1129]
As a user, I want to be able to use different components complementation with my webstudio build.

---

The Builder always uses React for its app. 

When it comes to the Canvas app, there are two modes: "edit mode" and "published mode." For these modes, we use React Vite/Snowpack, Remix, Nuxt (Vue), or Svelte-Kit app.

The components in the Canvas app can be made using React, Vue, or Svelte. 

Canvas app can override (wrap) some components depending on the framework or hosting abilities. 
_examples: wrap link component with Remix Link, add image resize i.e. cloudinary, cloudflare etc_

The react-sdk should synchronize between the Builder and the Canvas app, render components based on Builder data, and render CSS.

The Builder only needs component names, prop definitions, and icons. These are obtained from the SDK using the following code:

```ts
import {
  getComponentMeta,
  getComponentNames,
} from "@webstudio-is/react-sdk";
```

However, this means that a new component cannot be added without rebuilding the Builder.

Another issue is that the Builder has no knowledge of Canvas overrides since they are done at runtime in the Canvas app. To simplify this, we need to remove build-time dependencies and create a single source of truth for components, component props, and overrides. The Canvas app must inform the Builder about the list of components, component props, etc.

So we need.
- [x] - Make the Builder independent of build-time dependencies on components and their props
To achieve this, we need to make the Builder independent of build-time dependencies on components and their props. They must be atoms and synchronized via atoms from the Canvas. Once this is done, the component overrides will work in the Builder. 
What to do: 
- Remove the following imports from the Builder: getComponentMeta, getComponentPropsMeta, and getComponentNames.
- Synchronize meta, propsMeta, and component names through atoms and https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/app/shared/sync/sync-stores.ts (Canvas app -> Builder)

- [x] - Make react-sdk independent of components
Now we need to rebuild the SDK every time there is a change to a component. 

We can make the getComponentMeta, getComponentPropsMeta, getComponentNames, and getComponent methods init parameters of the React SDK, like init({components, definitions}). This way, the SDK does not need to include the components and can remain independent.

After this is done, the React SDK should only contain sync primitives and render primitives. The SDK should not include any components. Instead, it should rely on the components that are passed in through the init parameters.

By separating the components from the SDK, we can ensure that changes to the components do not require a rebuild of the React SDK. 

- [ ] - Move the canvas app into a standalone app.

After that we have.

Canvas app

- Import components and definitions
- initialize react-sdk with components and definitions

In case of component change, only the canvas app must be rebuilt 

Having that Canvas app for "publish" already has a build step i.e. saas, 
we can now think to create multiple "edit mode" canvas apps, with user-provided components.

**Comment by kof** [comment]
Goals

1. get rid of wrong types in components, which actually represent app-level properties
2. decouple builder from sdk

**Comment by TrySound** [comment]
Let's figure out canvas as separate app outside of this issue.
